static/app/components/modals/widgetViewerModal.tsx

```diff
@@ -35,83 +42,88 @@ type Props = ModalRenderProps &
     selection: PageFilters;
   };
 
-const TABLE_ITEM_LIMIT = 30;
-const FULL_TABLE_HEIGHT = 600;
-const HALF_TABLE_HEIGHT = 300;
+const FULL_TABLE_ITEM_LIMIT = 20;
+const HALF_TABLE_ITEM_LIMIT = 10;
 
 function WidgetViewerModal(props: Props) {
+  const {organization, widget, selection, location, Footer, Body, Header, onEdit} = props;
+  const eventView = eventViewFromWidget(
+    widget.title,
+    widget.queries[0],
+    selection,
+    widget.displayType
+  );
+  const isTableWidget = widget.displayType === DisplayType.TABLE;
   const renderWidgetViewer = () => {
-    const {organization, selection, widget, location} = props;
     const api = useApi();
-    switch (widget.displayType) {
-      case DisplayType.TABLE:
-        return (
-          <TableContainer height={FULL_TABLE_HEIGHT}>
+    const columnOrder = eventView.getColumns();
+    const columnSortBy = eventView.getSorts();
+    const Query: React.ComponentType<any> =
```

Comment:
in this case, would it be better if we did:

```suggestion
          {widget.widgetType === WidgetType.ISSUE ? (
            <IssueWidgetQueries {...commonQueryProps}>
              {({transformedResults, loading}) => {
                return (
                  <GridEditable
                    isLoading={loading}
                    data={transformedResults}
                    columnOrder={columnOrder}
                    columnSortBy={columnSortBy}
                    grid={{
                      renderHeadCell: renderGridHeaderCell(props),
                      renderBodyCell: renderGridBodyCell(props),
                    }}
                    location={location}
                  />
                );
              }}
            </IssueWidgetQueries>
          ) : (
            <WidgetQueries {...commonQueryProps}>
              {({tableResults, loading}) => {
                return (
                  <GridEditable
                    isLoading={loading}
                    data={tableResults?.[0]?.data ?? []}
                    columnOrder={columnOrder}
                    columnSortBy={columnSortBy}
                    grid={{
                      renderHeadCell: renderGridHeaderCell({
                        ...props,
                        tableData: tableResults?.[0],
                      }),
                      renderBodyCell: renderGridBodyCell({
                        ...props,
                        tableData: tableResults?.[0],
                      }),
                    }}
                    location={location}
                  />
                );
              }}
            </WidgetQueries>
          )}
```

Reply:
👍  updated! I initially wanted to reduce the repeat code lines but I think this works fine too.